### PR TITLE
Fix analyzing not caught some messages in default namespace

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Found here: https://github.com/istio/istio/pull/43626#pullrequestreview-1315927487
When there's any service with no selector in a namespace, `DeploymentRequiresServiceAssociated` message won't be reported because the no selector service will match all the deployments which actually have no associated service.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: releasenotes/notes/43652.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** analyzing not caught some messages when there's services with no selector exist in the analyzed namespace.
```
Comment (GregHanson) [inline releasenotes/notes/43652.yaml:6]:
```suggestion
    **Fixed** an bug in `istioctl analyze` where some messages are missed when there are services with no selector in the analyzed namespace.
```

**Review by GregHanson — COMMENTED** [review]
thanks for looking into this @hanxiaop

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
put a hold since there was open comment

**Comment by hanxiaop** [comment]
@howardjohn Thanks for the review! The comment has been resolved, and I'll remove the hold label.

**Pull request merged**